### PR TITLE
update xetla

### PR DESCRIPTION
**Pull request by zhewang1-intc**
support fpu int4 gemm
support activation shuffle
support nf4 weight_type

## Type of Change

feature or bug fix or documentation or others
API changed or not

## Description

detail description 
Issues: xxx

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -46,7 +46,9 @@ endif ()
 add_compile_options(-fsycl)
 add_link_options(-fsycl)
 if(UNIX)
-    add_compile_options(-fp-model=precise -Wall -Wextra -Werror)
```
Comment (DDEle) [inline CMakeLists.txt:49]: add back werror

File: include/experimental/group/gemm/compute_policy.hpp [hunk 2]
```diff
@@ -23,7 +23,7 @@
 
 namespace gpu::xetla::group {
 
-enum quant_mode : uint8_t { S4_ASYM, S4_FULLRANGE_NO_ZP };
+enum weight_dtype : uint8_t { S4_ASYM, S4_FULLRANGE_NO_ZP, NF4 };
```
Comment (airMeng) [inline include/experimental/group/gemm/compute_policy.hpp:26]: please consider merging S4_ASYM and S4_FULLRANG_NO_ZP

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 15">
<meta name=Originator content="Microsoft Word 15">
<link rel=File-List
href="file:///C:/Users/hengyume/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<link rel=themeData
href="file:///C:/Users/hengyume/AppData/Local/Temp/msohtmlclip1/01/clip_themedata.thmx">
<link rel=colorSchemeMapping
href="file:///C:/Users/hengyume/AppData/Local/Temp/msohtmlclip1/01/clip_colorschememapping.xml">

<style>

</style>

</head>

<body lang=EN-US style='tab-interval:36.0pt;word-wrap:break-word'>



  | Sym | storage | asym | stroage
-- | -- | -- | -- | --
FrontEnd | ZP=0, W4, Scale | ZP=8, W4'=(W4+8), Scale | ZP, W4, scale | ZP, W4, Scale
BackEnd(kernel) |   | (W4'  - ZP) * scale |   | (W4  - ZP) * scale




</body>

</html>

File: examples/CMakeLists.txt [hunk 3]
```diff
@@ -2,7 +2,7 @@ include_directories(${CMAKE_SOURCE_DIR}/include)
 include_directories(${CMAKE_SOURCE_DIR})
 
 # Creates a separate device code module for each SYCL* kernel
-# so that kernel for Dg2 and Xe will be JIT separately
+# so that kernel for gpu_arch::XeHpg and Xe will be JIT separately
```
Comment (DDEle) [inline examples/CMakeLists.txt:5]: unnecessay change

File: examples/09_gate_recurrent_unit/kernel_func.hpp [hunk 4]
```diff
@@ -113,7 +113,7 @@ struct gru_layer {
   using perf_tuning_knob =
       perf_tuning_knob_t<sg_tile_k, prefetch_distance, periodic_sync_interval>;
 
-  using compute_attr = xetla::group::compute_attr_t<T, T, Act_T>;
+  using compute_attr = group::compute_attr_t<T, T, Act_T>;
```
Comment (DDEle) [inline examples/09_gate_recurrent_unit/kernel_func.hpp:116]: will it compile? I added it as it confuses with sycl::group

File: tests/unit/reg_layout_conversion/kernel_func.hpp [hunk 5]
```diff
@@ -76,4 +76,4 @@ struct conversion_func {
     layout_convert(data_tile, linear_data_tile); // linear to tiled
     tile_store(data_tile, result_payload);
   }
-};
+};
```
Comment (DDEle) [inline tests/unit/reg_layout_conversion/kernel_func.hpp:79]: Please go thought the diff and revert unnecessary changes

File: include/group/cooperative_reduction.hpp [hunk 6]
```diff
@@ -180,8 +180,8 @@ class cooperative_reduce_t<
     nbarrier.wait();
 
     if (is_valid_post_process_wg()) {
-      // nbarrier.init_nbarrier(nbar_id, nbarrier_role::consumer);
-      // nbarrier.arrive();
+      nbarrier.init_nbarrier(nbar_id, nbarrier_role::consumer);
+      nbarrier.arrive();
```
Comment (DDEle) [inline include/group/cooperative_reduction.hpp:184]: Is it necessary? If so, where is its `wait()`?

File: include/group/gemm/impl/default_fpu_xe.hpp [hunk 7]
```diff
@@ -380,8 +380,6 @@ class gemm_t<
       SW_BARRIER();
       subgroup::tile_load<cache_hint::cached, cache_hint::cached>(
           matA, matA_payload);
-      if constexpr (!is_col_major_a)
-        reorder_matA(matA);
```
Comment (DDEle) [inline include/group/gemm/impl/default_fpu_xe.hpp:384]: I think you need to change accordingly in load_xe to keep the fmha correct

File: include/experimental/kernel/gemm/impl/int4_dequantize_kslicing_xe.hpp [hunk 8]
```diff
@@ -156,147 +157,12 @@ class gemm_universal_t<
...
-    matC_base_t matC_base;
-    /// @brief Is the base address of accumulation buffer.
-    acc_base_t acc_base;
-    /// @brief Is the base address of counter buffer.
-    cnt_base_t cnt_base;
-    /// @brief Is the epilogue arguments.
-    epilogue_args_t epilogue_args;
-
-    scale_base_t scale_base;
+  struct optional_argements_t {
```
Comment (DDEle) [inline include/experimental/kernel/gemm/impl/int4_dequantize_kslicing_xe.hpp:160]: typo
```suggestion
  struct optional_arguments_t {
```
